### PR TITLE
Added the reimplementation for the enumerable mehtods

### PR DESCRIPTION
**Pull request by AdfPizarro**
Reimplemented the enumerable methods 

 #my_each
 #my_each_with_index
 #my_select
 #my_all?
 #my_any?
 #my_none?
 #my_count
 #my_map
 #my_inject

Added test cases for each one

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @AdfPizarro and @gdumani 
### You have done really well so far :clap: However, you have a few issues to address before your project could be approved.
- [x] You need to update your readme file, make it descriptive enough
- [x] The following methods are not passing the tests:
- my_inject
- my_count
- my_all
- my_any
- my_none
### Make these changes and submit for another code review. Good luck.
- Happy coding!

By Raymond
slack @iamraymond

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    if acumulator.nil?
+      acumulator = to_a[0]
+      (1...length).my_each { |i| acumulator = acumulator.method(operator).call(to_a[i]) }
+    else
+      my_each { |i| acumulator = acumulator.method(operator).call(i) }
+    end
+  end
+  acumulator
+end
+# rubocop: enable Metrics/PerceivedComplexity
```
Comment (rOluochKe) [inline enumerables.rb]: ### The test case fails, producing an error
inject
- [ ] Remove the commented linting and do not disable them

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  true
+end
+
+def my_count
+  return enum_for(__callee__) unless block_given?
+
+  c = 0
+  my_each { |i| c += 1 if yield(i) }
+  c
+end
```
Comment (rOluochKe) [inline enumerables.rb]: ### Fix the failing error
count

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  my_each { |i| res << i if yield(i) }
+  res
+end
+
+def my_all
+  return enum_for(__callee__) unless block_given?
+
+  my_each { |i| return false unless yield(i) }
+  true
+end
```
Comment (rOluochKe) [inline enumerables.rb]: ### Fix the failing errors
all

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+def my_any
+  return enum_for(__callee__) unless block_given?
+
+  my_each { |i| return true if yield(i) }
+  false
+end
```
Comment (rOluochKe) [inline enumerables.rb]: ### Also fix this failing error 
any

File: enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+def my_none
+  return enum_for(__callee__) unless block_given?
+
+  my_each { |i| return false if yield(i) }
+  true
+end
```
Comment (rOluochKe) [inline enumerables.rb]: ### Fix the my_none method to pass the test
none

File: enumerables.rb (hunk 1 above)
Comment (AdfPizarro) [reply enumerables.rb]: Fixed updating length to size

File: enumerables.rb (hunk 3 above)
Comment (AdfPizarro) [reply enumerables.rb]: Updated the function name

File: enumerables.rb (hunk 2 above)
Comment (AdfPizarro) [reply enumerables.rb]: Updated function to receive parameters

File: enumerables.rb (hunk 5 above)
Comment (AdfPizarro) [reply enumerables.rb]: Updated method name

File: enumerables.rb (hunk 4 above)
Comment (AdfPizarro) [reply enumerables.rb]: Updated function name

File: enumerables.rb [hunk 6]
```diff
@@ -0,0 +1,168 @@
+module Enumerable
+  def my_each
+    return enum_for(__callee__) unless block_given?
+
+    size.times { |i| yield to_a[i] }
+  end
```
Comment (maelfosso) [inline enumerables.rb:6]:
```rb
2.6.5 :009 > array
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :010 > block = proc { |num| num < 4 }
 => #<Proc:0x00000000027ec8c0@(irb):10> 
2.6.5 :011 > array.my_each(&block)
 => 100 
2.6.5 :012 > array.each(&block)
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :013 > 
```

File: enumerables.rb [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+  def my_each_with_index
+    return enum_for(__callee__) unless block_given?
+
+    size.times { |i| yield(to_a[i], i) }
+  end
```
Comment (maelfosso) [inline enumerables.rb:12]:
```rb
2.6.5 :014 > array
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :015 > block = proc { |num| num < 4 }
 => #<Proc:0x00000000024fa1f8@(irb):15> 
2.6.5 :016 > array.my_each_with_index(&block)
 => 100 
2.6.5 :017 > array.each_with_index(&block)
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :018 > 
```

File: enumerables.rb [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    my_each { |i| res << i if yield(i) }
+    res
+  end
+
+  def my_all?
+    return enum_for(__callee__) unless block_given?
+
+    my_each { |i| return false unless yield(i) }
+    true
+  end
```
Comment (maelfosso) [inline enumerables.rb:27]:
```rb
2.6.5 :019 > true_array = [1, true, 'hi', []]
 => [1, true, "hi", []] 
2.6.5 :020 > true_array.my_all?
 => #<Enumerator: [1, true, "hi", []]:my_all?> 
2.6.5 :021 > true_array.all?
 => true 
2.6.5 :022 >
```

Comment (maelfosso) [reply enumerables.rb:27]:
```rb
2.6.5 :024 > array
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :025 > array.my_all?(Integer)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):25
        1: from enumerables.rb:22:in `my_all?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :026 > array.all?(Integer)
 => true 
2.6.5 :027 > 
```

Comment (maelfosso) [reply enumerables.rb:27]:
```rb
2.6.5 :028 > words = %w[dog door rod blade]
 => ["dog", "door", "rod", "blade"] 
2.6.5 :029 > words.my_all?(/d/)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):29
        1: from enumerables.rb:22:in `my_all?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :030 > words.all?(/d/)
 => true 
2.6.5 :031 > 
```

Comment (maelfosso) [reply enumerables.rb:27]:
```rb
2.6.5 :033 > array.my_all?(3)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):33
        1: from enumerables.rb:22:in `my_all?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :034 > array.all?(3)
 => false 
2.6.5 :035 >
```

File: enumerables.rb [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+
+  def my_any?
+    return enum_for(__callee__) unless block_given?
+
+    my_each { |i| return true if yield(i) }
+    false
+  end
```
Comment (maelfosso) [inline enumerables.rb:34]:
```rb
2.6.5 :036 > true_array = [nil, false, true, []]
 => [nil, false, true, []] 
2.6.5 :037 > true_array.any?
 => true 
2.6.5 :038 > true_array.my_any?
 => #<Enumerator: [nil, false, true, []]:my_any?> 
2.6.5 :039 >
```

Comment (maelfosso) [reply enumerables.rb:34]:
```rb
2.6.5 :041 > array
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :042 > array.my_any?(Integer)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):42
        1: from enumerables.rb:29:in `my_any?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :043 > array.any?(Integer)
 => true 
2.6.5 :044 >
```

Comment (maelfosso) [reply enumerables.rb:34]:
```rb
2.6.5 :045 > words
 => ["dog", "door", "rod", "blade"] 
2.6.5 :046 > words.any?(/z/)
 => false 
2.6.5 :047 > words.my_any?(/z/)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):47
        1: from enumerables.rb:29:in `my_any?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :048 > 
```

Comment (maelfosso) [reply enumerables.rb:34]:
```rb
2.6.5 :050 > words
 => ["dog", "door", "rod", "blade"] 
2.6.5 :051 > words.my_any?('cat')
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):51
        1: from enumerables.rb:29:in `my_any?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :052 > words.any?('cat')
 => false 
2.6.5 :053 >
```

File: enumerables.rb [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+
+  def my_none?
+    return enum_for(__callee__) unless block_given?
+
+    my_each { |i| return false if yield(i) }
+    true
+  end
```
Comment (maelfosso) [inline enumerables.rb:41]:
```rb
2.6.5 :055 > false_array = [nil, false, nil, false]
 => [nil, false, nil, false] 
2.6.5 :056 > false_array.my_none?
 => #<Enumerator: [nil, false, nil, false]:my_none?> 
2.6.5 :057 > false_array.none?
 => true 
2.6.5 :058 >
```

Comment (maelfosso) [reply enumerables.rb:41]:
```rb
2.6.5 :059 > array
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :060 > array.my_none?(String)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):60
        1: from enumerables.rb:36:in `my_none?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :061 > array.none?(String)
 => true 
2.6.5 :062 > 
```

Comment (maelfosso) [reply enumerables.rb:41]:
```rb
2.6.5 :063 > words
 => ["dog", "door", "rod", "blade"] 
2.6.5 :064 > words.my_none?(/z/)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):64
        1: from enumerables.rb:36:in `my_none?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :065 > words.none?(/z/)
 => true 
2.6.5 :066 >
```

Comment (maelfosso) [reply enumerables.rb:41]:
```rb
2.6.5 :068 > words
 => ["dog", "door", "rod", "blade"] 
2.6.5 :069 > words.my_none?(5)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):69
        1: from enumerables.rb:36:in `my_none?'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :070 > words.none?(5)
 => true 
2.6.5 :071 > 
```

File: enumerables.rb [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+    c = 0
+    if block_given?
+
+      my_each { |i| c += 1 if yield(i) }
+    else
+      my_each { |i| c += 1 if i == cond }
+
+    end
+    c
+  end
```
Comment (maelfosso) [inline enumerables.rb:53]:
```rb
2.6.5 :073 > array
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :074 > array.my_count
 => 0 
2.6.5 :075 > array.count
 => 100 
2.6.5 :076 >
```

File: enumerables.rb [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+
+  def my_map(&oper)
+    res = []
+    if block_given?
+      my_each { |i| res << yield(i) }
+    else
+      my_each { |i| res << oper.call(i) }
+    end
+    res
+  end
```
Comment (maelfosso) [inline enumerables.rb:63]:
```rb
2.6.5 :082 > array
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :083 > array.my_map
Traceback (most recent call last):
        9: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        8: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        7: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        6: from (irb):83
        5: from enumerables.rb:60:in `my_map'
        4: from enumerables.rb:5:in `my_each'
        3: from enumerables.rb:5:in `times'
        2: from enumerables.rb:5:in `block in my_each'
        1: from enumerables.rb:60:in `block in my_map'
NoMethodError (undefined method `call' for nil:NilClass)
2.6.5 :084 > array.map
 => #<Enumerator: [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1]:map> 
2.6.5 :085 > 
```

Comment (maelfosso) [reply enumerables.rb:63]:
```rb
2.6.5 :087 > my_proc = proc {|num| num > 10 }
 => #<Proc:0x00000000026d5388@(irb):87> 
2.6.5 :088 > array
 => [4, 1, 4, 4, 5, 3, 0, 4, 2, 8, 3, 4, 8, 0, 2, 0, 4, 2, 7, 0, 5, 4, 3, 3, 8, 8, 5, 3, 8, 8, 5, 4, 7, 6, 2, 1, 4, 1, 8, 1, 4, 1, 1, 5, 6, 8, 8, 5, 6, 0, 6, 8, 7, 6, 2, 0, 5, 5, 6, 6, 4, 5, 4, 2, 5, 0, 0, 6, 2, 5, 4, 4, 4, 7, 1, 0, 3, 8, 1, 3, 7, 8, 8, 4, 7, 5, 6, 0, 5, 2, 0, 5, 6, 3, 5, 2, 8, 7, 2, 1] 
2.6.5 :089 > array.map(my_proc)
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):89
        1: from (irb):89:in `map'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :090 > array.map(&my_proc)
 => [false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false] 
2.6.5 :091 > array.my_map(&my_proc)
 => [false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false, false] 
2.6.5 :092 > array.map(&my_proc) == array.my_map(&my_proc)
 => true 
2.6.5 :093 > 
2.6.5 :094 > array.my_map(my_proc) {|num| num < 10 }
Traceback (most recent call last):
        5: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /usr/share/rvm/rubies/ruby-2.6.5/bin/irb:23:in `load'
        3: from /usr/share/rvm/rubies/ruby-2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):94
        1: from enumerables.rb:55:in `my_map'
ArgumentError (wrong number of arguments (given 1, expected 0))
2.6.5 :095 > 
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

1. Please, the PR is from a feature branch to the `develop` branch. The `**master**` is not concerned about it. Take a look at [this](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/git_flow.png). Kindly create another PR according to that instruction. 

2. Read the comments below and check the differences between your methods output and the ones from inner Ruby methods.

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

File: enumerables.rb (hunk 6 above)
Comment (gdumani) [reply enumerables.rb:6]: Corrected return value to self

File: enumerables.rb (hunk 7 above)
Comment (gdumani) [reply enumerables.rb:12]: added self as return value

File: enumerables.rb (hunk 11 above)
Comment (gdumani) [reply enumerables.rb:53]: added option when no parameters are passed

File: enumerables.rb (hunk 8 above)
Comment (AdfPizarro) [reply enumerables.rb:27]: Updated to receive regexp and arrays

File: enumerables.rb (hunk 9 above)
Comment (AdfPizarro) [reply enumerables.rb:34]: Updated method

File: enumerables.rb (hunk 10 above)
Comment (AdfPizarro) [reply enumerables.rb:41]: Updated as a negation of my any

File: enumerables.rb (hunk 12 above)
Comment (AdfPizarro) [reply enumerables.rb:63]: Fixed when we updated the my_each method

**Comment by AdfPizarro** [comment]
Updated methods

Fixed the returns on my_each and my_each with index
Updated methods my_all, my_none, my_any to work with class names, regexp, and blocks
my_counter works now with no arguments 
Fixed my_map with the my_each update

**Pull request merged**